streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -2805,7 +2805,7 @@ <VT, VR> KStream<K, VR> leftJoin(final KTable<K, VT> table,
      * For each {@code KStream} record that finds a corresponding record in {@link GlobalKTable} the provided
      * {@link ValueJoiner} will be called to compute a value (with arbitrary type) for the result record.
      * The key of the result record is the same as the key of this {@code KStream}.
-     * If a {@code KStream} input record key or value is {@code null} the record will not be included in the join
+     * If a {@code KStream} input value is {@code null} the record will not be included in the join
      * operation and thus no output record will be added to the resulting {@code KStream}.
      * If {@code keyValueMapper} returns {@code null} implying no match exists, no output record will be added to the
      * resulting {@code KStream}.
```

Comment:
Should we merge both sentences:
```
If a {@code KStream} input record key or value is {@code null} the record will not be included in the join
     * If a {@code KStream} input value is {@code null} the record will not be included in the join
     * operation and thus no output record will be added to the resulting {@code KStream}.
     * If {@code keyValueMapper} returns {@code null} implying no match exists, no output record will be added to the
     * resulting {@code KStream}.
```

Into:
```
* If a {@code KStream} input value is {@code null} or if {@code keyValueMapper} returns {@code null} the record
* will not be included in the join operation and thus no output record will be added to the resulting {@code KStream}.
```

Same below? (Note: for left-join the JavaDocs about `null` returned from `keyValueMapper` seem to be wrong, too?

Reply:
I agree, it is more concise.
Regarding the left-join - yes, I adjusted that one too.